viper/src/main/scala/viper/api/SiliconVerifier.scala

```diff
@@ -7,17 +7,18 @@ import hre.ast.OriginFactory
 import hre.config.Configuration
 import viper.silver.plugin.PluginAwareReporter
 
-import scala.collection.JavaConverters._
+import scala.jdk.CollectionConverters._
 
 class SiliconVerifier[O](o:OriginFactory[O]) extends SilverImplementation[O](o) {
   override def createVerifier(z3Path: Path, z3Settings: Properties):viper.silver.verifier.Verifier = {
     val silicon = new viper.silicon.Silicon(PluginAwareReporter(HREViperReporter()), Seq("startedBy" -> "example", "fullCmd" -> "dummy"))
 
-    val z3Config = '"' + z3Settings.asScala.map{case (k, v) => s"$k=$v"}.mkString(" ") + '"'
+    val z3KeyValues = z3Settings.asScala.map{case (k, v) => s"$k=$v"}.mkString(" ")
 
     var siliconConfig = Seq(
       "--z3Exe", z3Path.toString,
-      "--z3ConfigArgs", z3Config,
+      // Ensure key-value pairs are wrapped in quotes without leading or trailing spaces
+      "--z3ConfigArgs", s""" "${z3KeyValues}" """.trim,
```

Comment:
what does this change do? as I see it:

* `s` means we have a format string on our hands
* triple quote means we can put literal quotes in the string
* the string thus is `<space><quote>z3KeyValues<quote><space>`
* we trim it, leaving us with `<quote>z3KeyValues<quote>`

why not just surround z3KeyValues in quotes normally? I think `mkString` accepts a start and end.

Reply:
Perfect! Although the "\"" still bother me, but I guess that's not avoidable.